pyanaconda/modules/storage/partitioning/interactive_utils.py

```diff
@@ -528,6 +667,56 @@ def collect_device_types(device, disks):
     return sorted(filter(devicefactory.is_supported_device_type, supported_types))
 
 
+def generate_device_info(storage, device):
+    """Generate a device info for the given device.
+
+    :param storage: an instance of Blivet
+    :param device: a device
+    :return: a device info
+    """
+    device_type = devicefactory.get_device_type(device)
+
+    dev_info = dict()
+    dev_info["min_luks_entropy"] = crypto.MIN_CREATE_ENTROPY
+    dev_info["device"] = device
+    dev_info["name"] = getattr(device.raw_device, "lvname", device.raw_device.name)
+    dev_info["size"] = device.size
+    dev_info["device_type"] = device_type
+    dev_info["fstype"] = device.format.type
+    dev_info["encrypted"] = isinstance(device, LUKSDevice)
+    dev_info["luks_version"] = get_device_luks_version(device)
+    dev_info["label"] = getattr(device.format, "label", "")
+    dev_info["mountpoint"] = getattr(device.format, "mountpoint") or None
+    dev_info["raid_level"] = get_device_raid_level(device)
```

Comment:
Please use enum or constants. It will help us to catch potential typos.

Reply:
The next step will be to replace these dictionaries with DBus structures, so it is not really worth it.

Reply:
It depends on how much you believe that this set of changes don't have a typo. But if you are planning to make the structure soon enough then I hope it will be fine.